### ISSUE TITLE
FlurlCall.Succeeded should null-check response, not request

### ISSUE THREAD

**Issue by kvpt** [tmenier/Flurl#573]
After upgrading Flurl 2.x to 3.0 some tests fail.

I identified the issue, it seems to be related to the SimulateTimeout.
If I call  SimulateTimeout and not add a RespondWith after, all next calls will throw FlurlHttpTimeoutException, that was not the case with Flurl 2.x.

Some tests (XUnit + Shouldly) to illustrate the issue.

```cs
[Fact]
public async Task TestOk()
{
    FlurlClient flurlClient = new FlurlClient("http://fakeurl.io/api");

    await Should.NotThrowAsync(async () => await flurlClient.Request().GetAsync());
}

[Fact]
public async Task TestTimeoutOk()
{
    HttpTest.SimulateTimeout();

    FlurlClient flurlClient = new FlurlClient("http://fakeurl.io/api");

    await Should.ThrowAsync<FlurlHttpTimeoutException>(async () => await flurlClient.Request().GetAsync());

    HttpTest.RespondWith();

    await Should.NotThrowAsync(async () => await flurlClient.Request().GetAsync());
}

[Fact]
public async Task TestTimeoutKo()
{
    HttpTest.SimulateTimeout();

    FlurlClient flurlClient = new FlurlClient("http://fakeurl.io/api");

    await Should.ThrowAsync<FlurlHttpTimeoutException>(async () => await flurlClient.Request().GetAsync());

    await Should.NotThrowAsync(async () => await flurlClient.Request().GetAsync());
}
```

I tried to workaround the issue by calling RespondWith just after SimulateTimeout, this fixed most of the tests.

But I encountered another issue.
> System.NullReferenceException
Object reference not set to an instance of an object.
   at Flurl.Http.FlurlCall.get_Succeeded()

From the code :
https://github.com/tmenier/Flurl/blob/ed1e7bdf41ca7ebc5928f24f94c71cdc6a351a62/src/Flurl.Http/FlurlCall.cs#L84

I think there is a typo, HttpRequestMessage need to be HttpResponseMessage

**Comment by tmenier** [comment]
> If I call SimulateTimeout and not add a RespondWith after, all next calls will throw FlurlHttpTimeoutException, that was not the case with Flurl 2.x.

That's by design. The change is detailed in #482. You have the right idea for a work-around, or if you only want to simulate a timeout for a specific endpoint and not others, you could try using `ForCallsTo` as described in #481.

> I think there is a typo, HttpRequestMessage need to be HttpResponseMessage

Yep, I think you're right about that one. I'll get that fixed. Thanks for pointing it out.

**Comment by kvpt** [comment]
I looked for the term timeout in the release note and don't found it, it was indeed the change #482.
So this is by design and it's ok for me, I will just adapt my code for these cases.
Thanks @tmenier

**Comment by tmenier** [comment]
Thanks, but you did still catch a small bug at the end there so I'll keep this open until that's fixed/released.

**Comment by stogle** [comment]
> Thanks, but you did still catch a small bug at the end there so I'll keep this open until that's fixed/released.

I encountered this `NullReferenceException` when calling `FlurlCall.Succeeded` in my error handler after a simulated timeout. I spotted the typo in the decompiled source and came here to find it was a known issue.

As a workaround, I've changed my code from
```
if (call.Succeeded)
```
to
```
// Workaround for https://github.com/tmenier/Flurl/issues/573
if (call.HttpResponseMessage != null && call.Succeeded)
```

**Comment by tmenier** [comment]
Fixed & released in 3.0.1. Your work-around is no longer needed.
